### PR TITLE
add auto-generated config reference documentation

### PR DESCRIPTION
**Pull request by dhellmann**
Use autodoc-pydantic to generate some
configuration reference info.

Add some example YAML to docstrings for the models
included.

Add a [docs] extra for dependencies for building
documentation.

### CODE REVIEW

**Review by shubhbapna — COMMENTED** [review]
LGTM but maybe @tiran or @rd4398 can approve it if there are no issues since I think they worked on this

**Review by rd4398 — APPROVED** [review]
LGTM! Will wait for others to review and approve

**Review by shubhbapna — APPROVED** [review]

**Comment by tiran** [comment]
Nice!

**Pull request merged**
